docs/TESTPLAN.md

```diff
@@ -0,0 +1,68 @@
+# Test Plan
+
+This document describes the testing architecture, goals and current status for
+the AWS Toolkit for VSCode.
+
+## Test goals
+
+In order of priority:
+
+1. Fast feedback from making a code change to seeing a result
+2. High confidence in releasing the Toolkit
+3. Avoid regressions
+
+Ratio of unit to integ tests: 90% unit tests, 10% system/acceptance tests.
+
+## Test categories
+
+The test suite has two categories of tests:
+
+-   Unit Tests: **fast** tests
+    -   Live in `src/test/`
+    -   The `vscode` API is available, but a full VSCode instance is not started.
```

Comment:
We do start a full instance of VS Code, and some tests even utilize it (e.g. some of the webview-related ones, specifically tied to token replacement...should these be considered integration tests?).

Reply:
Our current unit tests are a mix of unit tests and mildly functional tests. In an ideal world we would go through our existing tests and split them along a more pure definition of unit tests, and move anything that leverages vs code experimental instances into a different group (possibly integration)

Reply:
> In an ideal world we would go through our existing tests and split them along a more pure definition of unit tests, and move anything that leverages vs code experimental instances into a different group (possibly integration)

I am not sure that is ideal. Ideally all the tests run very fast, thus the distinction is meaningless. "Unit" is not technically the proper name for what we have, but I don't see why we would move anything.